lib/rb_inject.rb

```diff
@@ -0,0 +1,48 @@
+class Array
+
+  def inject_custom(accumulator = nil, operation = nil, &block)
+    if accumulator.nil? && operation.nil? && block.nil?
+      raise ArgumentError, "you must provide an operation or a block"
+    end
+
+    if operation && block
+      raise ArgumentError, "you must provide either an operation symbol or a block, not both"
```

Comment:
Metrics/LineLength: Line is too long. [93/80]<br>Style/SignalException: Use `fail` instead of `raise` to signal exceptions.